### ISSUE TITLE
"Unknown error occurred" when setting up via config flow

### ISSUE THREAD

**Issue by adorobis** [oischinger/ha_vicare#58]
When setting up via config flow for a discovered Vicare integration the menu shows "Unknown error occurred" message.
I have the current config_flow branch in custom_components and PyVicare version 2.10.0 library. Also tested with 2.11.1 with the same outcome. Message in the log:
```
2021-10-07 09:59:26 ERROR (MainThread) [aiohttp.server] Error handling request
Traceback (most recent call last):
  File "/usr/local/share/homeassistant/lib/python3.8/site-packages/aiohttp/web_protocol.py", line 422, in _handle_request
    resp = await self._request_handler(request)
  File "/usr/local/share/homeassistant/lib/python3.8/site-packages/aiohttp/web_app.py", line 499, in _handle
    resp = await handler(request)
  File "/usr/local/share/homeassistant/lib/python3.8/site-packages/aiohttp/web_middlewares.py", line 119, in impl
    return await handler(request)
  File "/usr/local/share/homeassistant/lib/python3.8/site-packages/homeassistant/components/http/security_filter.py", line 60, in security_filter_middleware
    return await handler(request)
  File "/usr/local/share/homeassistant/lib/python3.8/site-packages/homeassistant/components/http/forwarded.py", line 98, in forwarded_middleware
    return await handler(request)
  File "/usr/local/share/homeassistant/lib/python3.8/site-packages/homeassistant/components/http/request_context.py", line 24, in request_context_middleware
    return await handler(request)
  File "/usr/local/share/homeassistant/lib/python3.8/site-packages/homeassistant/components/http/ban.py", line 78, in ban_middleware
    return await handler(request)
  File "/usr/local/share/homeassistant/lib/python3.8/site-packages/homeassistant/components/http/auth.py", line 144, in auth_middleware
    return await handler(request)
  File "/usr/local/share/homeassistant/lib/python3.8/site-packages/homeassistant/components/http/view.py", line 136, in handle
    result = await result
  File "/usr/local/share/homeassistant/lib/python3.8/site-packages/homeassistant/components/config/config_entries.py", line 157, in post
    return await super().post(request, flow_id)
  File "/usr/local/share/homeassistant/lib/python3.8/site-packages/homeassistant/components/http/data_validator.py", line 62, in wrapper
    result = await method(view, request, *args, **kwargs)
  File "/usr/local/share/homeassistant/lib/python3.8/site-packages/homeassistant/helpers/data_entry_flow.py", line 111, in post
    result = await self._flow_mgr.async_configure(flow_id, data)
  File "/usr/local/share/homeassistant/lib/python3.8/site-packages/homeassistant/data_entry_flow.py", line 198, in async_configure
    result = await self._async_handle_step(flow, cur_step["step_id"], user_input)
  File "/usr/local/share/homeassistant/lib/python3.8/site-packages/homeassistant/data_entry_flow.py", line 256, in _async_handle_step
    result: FlowResult = await getattr(flow, method)(user_input)
  File "/home/homeassistant/homeassistant/custom_components/vicare/config_flow.py", line 59, in async_step_user
    await self.hass.async_add_executor_job(vicare_login, user_input)
  File "/usr/local/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
TypeError: vicare_login() missing 1 required positional argument: 'conf'
```

**Comment by adorobis** [comment]
Great, thanks. Indeed the config flow works ok now. There is a different issue though: the water_heater.vicare_water was working ok with the yesterday's version of vicare custom_component and today it is not allowing to control the DHW set temperature anymore. Not sure if this was fixed and now overwritten by the sync back from HA Core or if that was fixed in the PyVicare library.

**Comment by oischinger** [comment]
Water heater also works for me. You don't by chance have a log when it failed?

I believe we are on the finishing line when it comes to this PR. But we still need reviews on HA Core. Let's see how long it takes.

Then we can concentrate again on additional sensors.

**Comment by adorobis** [comment]
Nothing in the logs but I have just realized that it was late last night and my water heater was already not heating due to schedule. I've just tested it now and indeed the water heater target temperature is not available when the water heater is not heating due to schedule. So all good now!
But maybe we could somehow reflect that state in HA? Wouldn't it be better if the `water_heater` entity was in `off` state when it is not heating due to schedule set?
